### PR TITLE
Legiones Auxilia

### PR DESCRIPTION
**Pull request by Mayegelt**
Standard thing of needing line breaks removed.
This adds all the Legiones Auxilia rules as a sub entry to the Solar Auxilia entry. I had to add 18 new categories to the GST (1 for each legion). Then had to add to each of the legion roots in their own library to add that category to the legion IF it was also a Crusade detachment in the same was as the ZM tag is added.

This has not added the Overseer. But I have left room for it to be added easily to the unit root entries.

This also includes the changes made for the Sentinel stuff from my other pull request. https://github.com/BSData/horus-heresy/pull/3191

### CODE REVIEW

**Comment by Mayegelt** [comment]
Worked out I forgot to add rule link to the Legiones Auxilia (X) rule in the entries.

**Comment by Mayegelt** [comment]
@nstephenh if you could do your thing again to fix the spaces that would be great. Added in the thing I forgot to link in.

**Comment by nstephenh** [comment]
Done

**Comment by nstephenh** [comment]
Sons of horus are appearing in thousand sons: 
image

File: 2022 - Horus Heresy.gst [hunk 1]
```diff
@@ -1350,6 +1350,24 @@ If a Skimmer is Immobilised or Wrecked, its base is removed, if possible. If thi
     </categoryEntry>
     <categoryEntry name="Blackshields Primary Detachment" id="3410-6cb0-2f2b-a28a" hidden="false"/>
     <categoryEntry name="Blackshields Optional Detachment" id="e3b2-7cd8-eab7-92eb" hidden="false"/>
+    <categoryEntry name="Dark Angels - Crusade Detachment" id="df67-4304-4f8a-b4ea" hidden="false"/>
```
Comment (nstephenh) [inline 2022 - Horus Heresy.gst:1353]: I like this implementation. So you add this if there's a crusade force org of the appropriate type, and select it in solar if true.

File: 2022 - LA - Blood Angels.cat [hunk 2]
```diff
@@ -3123,7 +3128,7 @@ Sanguinius may still Run on a turn when his wings have been activated. When maki
         <selectionEntry type="model" import="true" name="Sanguinius" hidden="false" id="c604-746e-d68b-3541">
           <categoryLinks>
             <categoryLink targetId="4f07-3d45-4f28-a0c6" id="2bb1-2477-683f-3f11" primary="false" name="Independent Character"/>
-            <categoryLink targetId="ad5f-31db-8bc7-5c46" id="d97c-b448-4ada-821f" name="Primarch:" primary="false"/>
+            <categoryLink targetId="ad5f-31db-8bc7-5c46" id="d97c-b448-4ada-821f" name="Primarch Unit Type" primary="false"/>
```
Comment (nstephenh) [inline 2022 - LA - Blood Angels.cat:3131]: This is probably just new recruit fixing link names it hadn't fixed before.

**Review by nstephenh — APPROVED** [review]

**Pull request merged**